### PR TITLE
GG-33002 Fix DistributedMetaStoragePersistentTest.testDeactivateActivate flaky test.

### PR DESCRIPTION
**Pull request by Berkof**
Cluster state checks were added into IgniteStatisticsManagerImpl and IgniteStatisticsConfigurationManager.

### CODE REVIEW

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java [hunk 1]
```diff
@@ -429,6 +430,10 @@ private void checkStatisticsSupport(String op) throws IgniteCheckedException {
            throw new IgniteCheckedException(String.format(
                "Unable to perform %s due to not all server nodes supports STATISTICS_COLLECTION feature.", op));
        }
+
+       if (ctx.state().clusterState().state() != ClusterState.ACTIVE)
+           throw new IgniteException(String.format(
+               "Unable to perform %s due to cluster state [state=%s]", op, ctx.state().clusterState().state()));
```
Comment (AMashenkov) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java:436]: Why not MessageFormat.format?

**Review by AMashenkov — APPROVED** [review]

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java [hunk 2]
```diff
@@ -334,6 +339,12 @@ public void stop() {
 
     /** */
     private void scanAndCheckLocalStatistics(AffinityTopologyVersion topVer) {
+        if (!active) {
+            if (log.isDebugEnabled())
+                log.debug("Ignore statistics collection due to inactive cluster state.");
+
+            return;
+        }
```
Comment (AMashenkov) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java:347]: Why is async task below not synchronized with deActivation?

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java (hunk 1 above)
Comment (Berkof) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java:436]: String.format used more than 100 times in codebase while MessageFormat.format - only 9. Is there are some advantages in your method?

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java (hunk 2 above)
Comment (Berkof) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java:347]: Move check into async task.

**Pull request merged**